src/main/java/fi/aalto/cs/apluscourses/ui/courseproject/CourseProjectDialog.java

```diff
@@ -0,0 +1,95 @@
+package fi.aalto.cs.apluscourses.ui.courseproject;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.DialogWrapper;
+import com.intellij.openapi.ui.Messages;
+import java.awt.event.ItemEvent;
+import javax.swing.Action;
+import javax.swing.JCheckBox;
+import javax.swing.JComponent;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.SwingConstants;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class CourseProjectDialog extends DialogWrapper {
```

Comment:
Swing components are totally unit-testable, shall this be tried?

Reply:
I'll try adding some tests :+1: